### DISCUSSION TITLE
CTRE paid 6.0 version?

### DISCUSSION THREAD

**Discussion by martygubow** [HuskieRobotics/3061-lib#96] [Q&A]
Does the 3061-lib require the CTRE paid 6.0 version. I get a warning that I have unlicensed devices.   I tried returning  getPhoenix6Licensed

**Comment by gcschmit** [comment]
It does not require the Phoenix Pro. Does your robot's config file return false for the getPhoenix6Licensed method? If so, do you have a link to your GitHub so I could take a closer look.

**Reply by martygubow** [reply]
I am running on a version one Rio with the built in CAN bus.   I was also getting some strange CAN bus errors such as no support for multi-signal  synchronization.  I decided to revert back to Phoenix 5  and my trusted code based on last season it uses  one of the first versions of 3061-lib.   I spent may hours trying to get the new version   of 3061-lib to work, but I may be CAN bus limited along with other issues.  I will need to transition in the next off season. The old version I am running has helped our team a ton.. Thank you!..... 

I could use a big favor.. I am trying to implement AutoBuilder.configureHolonomic in the old version of  Drivetrain.java.   I am having some issues.   The code is at:  https://github.com/FRC7224/FRC2024B   Any help would be appreciated!  
Thank you!   

Marty
Team 7224

**Comment by gcschmit** [comment]
Marty,

Sorry that you ran into this issue. I created an issue for us to debug it and ensure Phoenix Pro is not required. We must have missed something.

The new PathPlanner requires a few changes. I’ll attach a version of Drivetrain.java that should work. I tried to minimize the code changes. Here is what I think is needed:

create a new getRobotRelativeSpeeds (you had a method to do this but it isn’t compiling since it is defined in the constructor and is missing a method name)
this method should return the actual ChassisSpeeds and not the requested chassis speeds, which is why the ChassisSpeeds are constructed from the current SwerveModuleState objects
change the driveRobotRelative method to control the swerve modules directly and not call the existing drive method (also needs to be moved outside of the constructor)
the existing drive method doesn’t support overriding field relative or specifying closed loop velocity control
probably easier to have a method that programs the swerve modules directly

Here is the edited [Drivetrain.java](https://gist.github.com/gcschmit/6918edb630c8d26f893b0a6b3a3a734c).

Sorry again for the trouble :(

geoff


￼

**Reply by martygubow** [reply]
Geoff, 

Thank you!  Using the 3061-lib has been great! When I have more time, I will get ported to the new version. I really appreciate your help!

Marty
